src/git_changelog/cli.py

```diff
@@ -468,30 +570,39 @@ def main(args: list[str] | None = None) -> int:
     parser = get_parser()
     opts = parser.parse_args(args=args)
 
-    if opts.release_notes:
+    # Determine which arguments were explicitly set with the CLI
+    sentinel = _Sentinel()
+    sentinel_ns = argparse.Namespace(**{key: sentinel for key in vars(opts)})
+    parser.parse_args(namespace=sentinel_ns)
+    explicit_opts_dict = {
+        key: value for key, value in vars(sentinel_ns).items()
+        if value is not sentinel
+    }
+
+    config_file = explicit_opts_dict.pop("config_file", DEFAULT_CONFIG_FILES)
+    if str(config_file).strip().lower() in ("no", "none", "off", "false", "0", ""):
+        config_file = None
+    elif str(config_file).strip().lower() in ("yes", "default", "on", "true", "1"):
+        config_file = DEFAULT_CONFIG_FILES
+
+    settings = read_config(config_file)
+
+    # CLI arguments override the config file settings
+    settings.update(explicit_opts_dict)
+
+    if settings.pop("release_notes"):
         output_release_notes(
-            input_file=opts.input,
-            version_regex=opts.version_regex,
-            marker_line=opts.marker_line,
-            output_file=opts.output,
+            input_file=settings["input"],
+            version_regex=settings["version_regex"],
+            marker_line=settings["marker_line"],
+            output_file=settings["output"],
         )
         return 0
 
+    # --input is not necessary anymore
+    settings.pop("input", None)
```

Comment:
Not a big fan of having to do this. I think I'd prefer no `pop` and explicitly passing each value, even if verbose :slightly_smiling_face:

Reply:
If the number of options grows over time, the code itemizing the dictionary of settings is more reliable because you will not leave any options behind by mistake. However, if the dictionary keeps (does not pop) accidentally, you will immediately know because the build function will complain.

I have no attachment to this implementation, but this is how some other libraries handle keyword arguments when functions are wrapped around other lower-level functions (e.g., scikit-learn, and even more common with layers around matplotlib such as seaborn or nilearn).

Since Python will check you are not passing extra arguments unsupported by the function, there is no gain in passing arguments explicitly. However, popping ensures you track what is in that dictionary.

If this doesn't sound convincing enough, happy to change.

Reply:
Convinced by this: 

> leave any options behind by mistake

Thanks!